### PR TITLE
JNG-5928 groups added to views

### PR DESCRIPTION
**Pull request by noherczeg**
<table>
<td>
  <a href="https://blackbelt.atlassian.net/browse/JNG-5928" title="JNG-5928" target="_blank">StoryJNG-5928</a>  Groups to views in JSL
  </td></table>
  <br />

### CODE REVIEW

**Review by bencegelei — APPROVED** [review]

**Review by gaborflorian — APPROVED** [review]

File: model/src/main/java/hu/blackbelt/judo/meta/jsl/validation/JslDslValidator.xtend [hunk 1]
```diff
@@ -818,15 +818,15 @@ class JslDslValidator extends AbstractJslDslValidator {
...
-    }
+//    @Check
+//    def checkForDuplicateNameForEntityMemberDeclaration(EntityMemberDeclaration member) {
+//        if (member instanceof Named && (member.eContainer as EntityDeclaration).getMemberNames(member).map[n | n.toLowerCase].contains(member.name.toLowerCase)) {
+//            error("Duplicate member declaration: '" + member.name + "'",
+//                member.nameAttribute,
+//                DUPLICATE_MEMBER_NAME,
+//                member.name)
+//        }
+//    }
```
Comment (robertcsakany) [inline model/src/main/java/hu/blackbelt/judo/meta/jsl/validation/JslDslValidator.xtend:829]: What is the reason of commented out check? Where is the alternative?

File: model/src/main/java/hu/blackbelt/judo/meta/jsl/validation/JslDslValidator.xtend [hunk 2]
```diff
@@ -886,19 +886,34 @@ class JslDslValidator extends AbstractJslDslValidator {
...
+//        if (declaration instanceof FunctionDeclaration) return;
+//        if (declaration instanceof LambdaDeclaration) return;
+//
+//        if ((declaration.eContainer as ModelDeclaration).getDeclarationNames(declaration).map[n | n.toLowerCase].contains(declaration.name.toLowerCase)) {
+//            error("Duplicate declaration name: '" + declaration.name + "'",
+//                declaration.nameAttribute,
+//                DUPLICATE_DECLARATION_NAME,
+//                declaration.name)
+//        }
+//    }
```
Comment (robertcsakany) [inline model/src/main/java/hu/blackbelt/judo/meta/jsl/validation/JslDslValidator.xtend:902]: What is the reason of commented out check? Where is the alternative?

File: model/src/main/java/hu/blackbelt/judo/meta/jsl/validation/JslDslValidator.xtend [hunk 3]
```diff
@@ -1480,36 +1495,26 @@ class JslDslValidator extends AbstractJslDslValidator {
...
+//		
+//		val members = transfer.eAllContents.filter[d | d instanceof TransferMemberDeclaration].map[d | d as TransferMemberDeclaration].toList
+//
+//        if (members.filter[m | m.name.toLowerCase.equals(member.name.toLowerCase)].size > 1) {
+//            error("Duplicate member declaration: '" + member.name + "'.",
+//                member.nameAttribute,
+//                DUPLICATE_MEMBER_NAME,
+//                member.name)
+//        }
+//    }
```
Comment (robertcsakany) [inline model/src/main/java/hu/blackbelt/judo/meta/jsl/validation/JslDslValidator.xtend:1517]: What is the reason of commented out check? Where is the alternative?

File: model-test/src/test/java/hu/blackbelt/judo/meta/jsl/runtime/NameDuplicationDetectionTests.xtend [hunk 4]
```diff
@@ -40,18 +40,16 @@ class NameDuplicationDetectionTests {
...
             entity B {
-                relation A a opposite b;
+                relation A a opposite-add:b;
             }
 
         '''.parse => [
-            assertDuplicateMemberName("Duplicate member declaration: 'b'", JsldslPackage::eINSTANCE.entityFieldDeclaration)
-            assertDuplicateMemberName("Duplicate member declaration: 'b'", JsldslPackage::eINSTANCE.entityRelationDeclaration)
-        ]
+            	m | m.assertError(JsldslPackage::eINSTANCE.entityRelationOppositeInjected, JslDslValidator.DUPLICATE_MEMBER_NAME)
```
Comment (robertcsakany) [inline model-test/src/test/java/hu/blackbelt/judo/meta/jsl/runtime/NameDuplicationDetectionTests.xtend:51]: This change means different test scenario! When testg for opposite required, create new tests for that!

**Review by robertcsakany — CHANGES_REQUESTED** [review]

File: model/src/main/java/hu/blackbelt/judo/meta/jsl/validation/JslDslValidator.xtend (hunk 1 above)
Comment (gprivi) [reply model/src/main/java/hu/blackbelt/judo/meta/jsl/validation/JslDslValidator.xtend:829]: The "def checkForDuplicateName(Named named)" changed to check any Named declaration.
The commented out code is temporary and will be removed later.

File: model/src/main/java/hu/blackbelt/judo/meta/jsl/validation/JslDslValidator.xtend (hunk 2 above)
Comment (gprivi) [reply model/src/main/java/hu/blackbelt/judo/meta/jsl/validation/JslDslValidator.xtend:902]: The "def checkForDuplicateName(Named named)" changed to check any Named declaration.
The commented out code is temporary and will be removed later.

File: model/src/main/java/hu/blackbelt/judo/meta/jsl/validation/JslDslValidator.xtend (hunk 3 above)
Comment (gprivi) [reply model/src/main/java/hu/blackbelt/judo/meta/jsl/validation/JslDslValidator.xtend:1517]: The "def checkForDuplicateName(Named named)" changed to check any Named declaration.
The commented out code is temporary and will be removed later.

**Pull request merged**